src/main/python/community-detector/community_detector.py

```diff
@@ -0,0 +1,141 @@
+from collections import defaultdict
+from itertools import chain
+import logging
+from igraph import Graph
+import numpy
+
+
+def detect_communities(cc,
+                       buckets_matrix,
+                       edges="linear",
+                       algorithm="walktrap",
+                       algorithm_params={}):
+    """
+    Runs Community Detection analysis on the given connected components.
+    :param cc: numpy.ndarray with the connected components
```

Comment:
according to code looks like it doesn't have to be `numpy.ndarray`. But if I'm wrong, please specify shape.